internal/typeinfo/valuelocator.go

```diff
@@ -172,6 +174,44 @@ func (f *structField) LocateScanTarget(typeToValue map[reflect.Type]reflect.Valu
 	return val.Addr().Interface(), nil, nil
 }
 
+// slice represents a slice input.
+type slice struct {
+	sliceType reflect.Type
+}
+
+// Desc returns a natural language description of the slice for use in error
+// messages.
+func (s *slice) Desc() string {
+	return fmt.Sprintf("slice %q", s.sliceType.Name())
+}
+
+// Identifier returns a string that uniquely identifies the slice type in the
+// context of the query.
+func (s *slice) Identifier() string {
+	return s.sliceType.Name() + "[:]"
+}
+
+// ArgType is the type of the slice input to extract query parameters from.
+func (s *slice) ArgType() reflect.Type {
+	return s.sliceType
+}
+
+// LocateParams locates the slice argument assosiated with the slice
+// ValueLocator in typeToValue and returns the reflect.Value objects generated
+// by reflecting on the elements of the slice.
+func (s *slice) LocateParams(typeToValue map[reflect.Type]reflect.Value) ([]reflect.Value, error) {
+	sv, ok := typeToValue[s.sliceType]
+	if !ok {
+		return nil, valueNotFoundError(typeToValue, s.sliceType)
+	}
+
+	params := []reflect.Value{}
+	for i := 0; i < sv.Len(); i++ {
+		params = append(params, sv.Index(i))
+	}
+	return params, nil
+}
+
 // locateValue locates the value corresponding to the given type in the
 // typeToValueMap and returns an error message if it cannot be found.
 func locateValue(typeToValue map[reflect.Type]reflect.Value, typ reflect.Type) (reflect.Value, error) {
```

Comment:
You've started something here, and left it unfinished. It isn't used.

Reply:
Ah yea, I started writing locateValue to lookup values in the `typeToValue` map but decided it was a unnecessary overhead and it could just be done inline, must have forgotten to remove it.